### PR TITLE
Added SelectionData into marshalers, fixed SetData and GetData crashing

### PR DESCRIPTION
**Pull request by diamondburned**
This PR fixes the two below crashes, as well as adding SelectionData into the list of known Glib marshalers for use in Connect callback arguments.

**Note:** This PR breaks existing code using `gtk.SetData(uintptr(unsafe.Pointer(v)), ...)`, which may or may not be a good thing (as the use of `uintptr` in the API should be discouraged).

The crash for SetData:
```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc8 pc=0x7f3fa47abba0]

runtime stack:
runtime.throw(0xd67467, 0x2a)
	/nix/store/1r61f7iwqfb6dm1krqpj9cjfifjnfmjx-go-1.14.2/share/go/src/runtime/panic.go:1116 +0x72
runtime.sigpanic()
	/nix/store/1r61f7iwqfb6dm1krqpj9cjfifjnfmjx-go-1.14.2/share/go/src/runtime/signal_unix.go:679 +0x46a

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0xb7b080, 0xc0004616e0, 0xc00074c420)
	/nix/store/1r61f7iwqfb6dm1krqpj9cjfifjnfmjx-go-1.14.2/share/go/src/runtime/cgocall.go:133 +0x5b fp=0xc0004616b0 sp=0xc000461678 pc=0x48560b
github.com/gotk3/gotk3/gtk._Cfunc_gtk_selection_data_set(0xa8, 0x82, 0x8, 0xc00002bc40, 0xc000000011)
	_cgo_gotypes.go:22496 +0x45 fp=0xc0004616e0 sp=0xc0004616b0 pc=0x605c75
github.com/gotk3/gotk3/gtk.(*SelectionData).SetData.func1(0x7ffddb25e9b0, 0x82, 0xc00002bc40, 0x11, 0x20)
	/home/diamond/Scripts/gotk3/gtk/gtk.go:7516 +0xc1 fp=0xc000461728 sp=0xc0004616e0 pc=0x69c291
github.com/gotk3/gotk3/gtk.(*SelectionData).SetData(0x7ffddb25e9b0, 0x82, 0xc00002bc40, 0x11, 0x20)
	/home/diamond/Scripts/gotk3/gtk/gtk.go:7516 +0x53 fp=0xc000461760 sp=0xc000461728 pc=0x64c163
```

The crash for GetData:

```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd0 pc=0x7f3ec0e36b05]

runtime stack:
runtime.throw(0xd67467, 0x2a)
	/nix/store/1r61f7iwqfb6dm1krqpj9cjfifjnfmjx-go-1.14.2/share/go/src/runtime/panic.go:1116 +0x72
runtime.sigpanic()
	/nix/store/1r61f7iwqfb6dm1krqpj9cjfifjnfmjx-go-1.14.2/share/go/src/runtime/signal_unix.go:679 +0x46a

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0xb7afa0, 0xc0001596c8, 0x30)
	/nix/store/1r61f7iwqfb6dm1krqpj9cjfifjnfmjx-go-1.14.2/share/go/src/runtime/cgocall.go:133 +0x5b fp=0xc000159698 sp=0xc000159660 pc=0x48560b
github.com/gotk3/gotk3/gtk._Cfunc_gtk_selection_data_get_data_with_length(0xa8, 0xc000574058, 0x0)
	_cgo_gotypes.go:22444 +0x4e fp=0xc0001596c8 sp=0xc000159698 pc=0x605aee
github.com/gotk3/gotk3/gtk.(*SelectionData).GetData.func1(0x7ffe77a8c000, 0xc000574058, 0xc0000f6540)
	/home/diamond/Scripts/gotk3/gtk/gtk.go:7487 +0x64 fp=0xc000159700 sp=0xc0001596c8 pc=0x69c1a4
github.com/gotk3/gotk3/gtk.(*SelectionData).GetData(0x7ffe77a8c000, 0x0, 0x0, 0x0)
	/home/diamond/Scripts/gotk3/gtk/gtk.go:7487 +0x56 fp=0xc000159730 sp=0xc000159700 pc=0x64c0d6
```

### CODE REVIEW

**Comment by diamondburned** [comment]
Can we have this merged anyway? I feel like a small breaking change like this would be very easy to fix. It would also get rid of the previous function that uses `uintptr`, which is very unsafe.

File: gtk/gtk.go [hunk 1]
```diff
@@ -7472,7 +7472,11 @@ func (v *SelectionData) native() *C.GtkSelectionData {
 	if v == nil {
 		return nil
 	}
-	return v.GtkSelectionData
+
+	// I don't understand why we need this, but we do.
```
Comment (MJacred) [inline gtk/gtk.go:7476]: `v` is already of type `*C.GtkSelectionData`. This cast is not needed.

File: gtk/gtk.go [hunk 2]
```diff
@@ -7492,28 +7496,13 @@ func (v *SelectionData) GetData() (data []byte) {
...
-
-	return byteData
-}
-
-//for "drag-data-get"
-func SetData(pointer uintptr, atom gdk.Atom, data []byte) {
-	c := (*C.GValue)(unsafe.Pointer(pointer))
-	p := (*C.GtkSelectionData)(unsafe.Pointer(c))
-	C.gtk_selection_data_set(p, C.GdkAtom(unsafe.Pointer(atom)), C.gint(8), (*C.guchar)(unsafe.Pointer(&data[0])), C.gint(len(data)))
+// SetData is a wrapper around gtk_selection_data_set_data_with_length.
```
Comment (MJacred) [inline gtk/gtk.go:7499]: `gtk_selection_data_set_data_with_length` is not called here. it's `gtk_selection_data_set_data`

**Comment by MJacred** [comment]
FYI: Can't say anything regarding the funcs taking a `uintptr` parameter. They are convenience funcs added for Drag & Drop functionality (at least when you look at the commits which introduced them...).

File: gtk/gtk.go (hunk 1 above)
Comment (diamondburned) [reply gtk/gtk.go:7476]: Without this cast, the function would crash the application. The original code also had it.

**Comment by diamondburned** [comment]
> FYI: Can't say anything regarding the funcs taking a `uintptr` parameter. They are convenience funcs added for Drag & Drop functionality (at least when you look at the commits which introduced them...).

I made this PR precisely to implement Drag & Drop functionality. Taking in `uintptr` would require unsafe casting, so I'm not sure what you mean.

**Pull request merged**